perfkitbenchmarker/windows_benchmarks/geekbench_benchmark.py

```diff
@@ -0,0 +1,164 @@
+              For instance, if a throughput value is provided, the corresponding score is also included.
+
+    Example Output:
+        [
+            {
+                "metric_name": "Single-Core Score",
+                "metric_value": 1795,
+                "metric_unit": "points",
+                "metric_metadata": {"category": "Single-Core"}
+            },
+            {
+                "metric_name": "Single-Core File Compression",
+                "metric_value": 269.3,
+                "metric_unit": "MB/sec",
+                "metric_metadata": {
+                    "category": "Single-Core",
+                    "test": "File Compression",
+                    "score": 1875
+                }
+            }
+        ]
+    """
+
+    # Initialize a list to store the parsed samples
+    samples = []
+
+    # Track the current category (Single-Core, Multi-Core, or OpenCL)
+    current_category = None
+    current_metric_name = None
+    last_score = None 
+
+    # Split the output into lines for easier processing
+    lines = geekbench_output.splitlines()
+
+    for line in lines:
+        line = line.strip() 
+        # Detect category headers
+        if "Single-Core Performance" in line:
+            current_category = "Single-Core"
+        elif "Multi-Core Performance" in line:
+            current_category = "Multi-Core"
+        elif "OpenCL Performance" in line:
+            current_category = "OpenCL"
+        
+        # Detect overall score lines, ensuring current_category is not None
+        elif "Score" in line and current_category:
+            # Parse overall score based on the current category
+            try:
+                score = int(line.split()[-1])
+                samples.append({
+                    "metric_name": f"{current_category} Score",
+                    "metric_value": score,
+                    "metric_unit": "points",
+                    "metric_metadata": {"category": current_category}
+                })
+            except ValueError:
+                # Handle the case where score parsing fails
+                continue
+
+        # Detect specific test names within a category
+        elif line.strip() and line.split()[0].isalpha():
+            current_metric_name = line.strip()
+        
+        # Detect score line before throughput, storing score for metadata
+        elif current_metric_name and line.strip().isdigit():
+            last_score = int(line.strip())
+        
+        # Detect throughput values with units (e.g., 269.3 MB/sec)
+        elif current_metric_name and line.strip():
+            parts = line.strip().split()
+            try:
+                value = float(parts[0])  # First part is the numeric value
+                unit = ' '.join(parts[1:]) if len(parts) > 1 else 'points'  # Remaining part is the unit
+
+                # Add the parsed data as a sample, including the last_score in metadata
+                samples.append({
```

Comment:
Use a sample.Sample object.
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/sample.py#L121

Reply:
Got it, thanks!